libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -306,19 +311,68 @@ void AP_Camera_Backend::check_feedback()
     }
 }
 
-void AP_Camera_Backend::prep_mavlink_msg_camera_feedback(uint64_t timestamp_us)
+void AP_Camera_Backend::prep_capture_feedback(uint64_t timestamp_us)
 {
     const AP_AHRS &ahrs = AP::ahrs();
-    if (!ahrs.get_location(camera_feedback.location)) {
-        // completely ignore this failure!  AHRS will provide its best guess.
+    Location camera_location;
+    // completely ignore this failure!  AHRS will provide its best guess.
+    UNUSED_RESULT(ahrs.get_location(camera_feedback.location));
+
+    prep_mavlink_msg_camera_image_captured(timestamp_us, camera_location);
+    GCS_SEND_MESSAGE(MSG_CAMERA_IMAGE_CAPTURED);
+
+    if (_frontend.use_deprecated_feedback_msg()) {
+        prep_mavlink_msg_camera_feedback(timestamp_us, camera_location);
+        GCS_SEND_MESSAGE(MSG_CAMERA_FEEDBACK);
     }
+}
+
+void AP_Camera_Backend::prep_mavlink_msg_camera_feedback(uint64_t timestamp_us, Location &camera_location)
```

Comment:
```suggestion
#if AP_CAMERA_MAVLINK_FEEDBACK_MESSAGE_ENABLED
void AP_Camera_Backend::prep_mavlink_msg_camera_feedback(uint64_t timestamp_us, Location &camera_location)
```

... and add to _config.h

Reply:
Done.